### PR TITLE
SWR API calls and Sign modal loading

### PR DESCRIPTION
**Pull request by razvantomegea**
### Feature
Added SWR functions for API calls and improved sign modal initial loading

### Contains breaking changes
[x] No

[] Yes

### Updated CHANGELOG
[x] Yes

### Testing
[x] User testing
[] Unit tests

### CODE REVIEW

**Review by MiroMargineanu — APPROVED** [review]

**Review by MiroMargineanu — APPROVED** [review]

File: src/UI/SignTransactionsModals/SignWithDeviceModal/SignStep.tsx [hunk 1]
```diff
@@ -89,9 +90,14 @@ const SignStepComponent = (props: SignStepType & WithStylesImportType) => {
 
   const onSubmit = async () => {
     await onSignTransaction();
+
     if (signLastTransaction && GuardianScreen) {
       return setShowGuardianScreen(true);
     }
+
+    if (signLastTransaction && handleSubmit) {
```
Comment (arhtudormorar) [inline src/UI/SignTransactionsModals/SignWithDeviceModal/SignStep.tsx:98]: ?

File: src/apiCalls/configuration/useGetServerConfigurationFromApi.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    return null;
+  }
+
+  if (error) {
+    console.error('Error fetching server configuration:', error);
+  }
+
+  const egldDenomination = 'egldDenomination';
+
+  if (egldDenomination in data) {
```
Comment (arhtudormorar) [inline src/apiCalls/configuration/useGetServerConfigurationFromApi.ts:20]: early return?

File: src/hooks/transactions/useSignTransactionsCommonData.tsx [hunk 3]
```diff
@@ -39,7 +38,7 @@ export const useSignTransactionsCommonData = () => {
     const transactionsWithFixedNonce = transactionsToSign?.transactions ?? [];
 
     if (hasTransactionsToSign) {
-      const transactionsWithIncrementalNonces = await setTransactionNonces(
```
[the remark this answers is not in the document]
Comment (razvantomegea) [reply src/hooks/transactions/useSignTransactionsCommonData.tsx:42]: It is no longer async

File: src/UI/SignTransactionsModals/SignWithDeviceModal/SignStep.tsx (hunk 1 above)
Comment (razvantomegea) [reply src/UI/SignTransactionsModals/SignWithDeviceModal/SignStep.tsx:98]: Useful if we want to do an action on submit

**Pull request merged**